### PR TITLE
Migrates anon address to uphold wallet object

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/10397

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
plan 1
- enable rewards
- connect kyc wallet
- make sure that you see anon_address filled in uphold object in Preferences file

plan 2
- start on master
- enable rewards 
- connect kyc wallet
- make sure that you see `uphold_anon_address` in Preferences file
- close browser
- switch to this PR and start the browser
- make sure that you see anon_address filled in uphold object and no more `uphold_anon_address` in Preferences file

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/state/state_migration_v3.cc [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "base/base64.h"
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/state/state_migration_v3.cc:6]: remove

File: vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.h [hunk 2]
```diff
@@ -86,13 +86,11 @@ class Wallet {
 
   std::string GetClaimPayload(
       const std::string user_funds,
-      const std::string new_address,
+      const std::string address,
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.h:89]: All of these parameters should probably be const references.

File: vendor/bat-native-ledger/src/bat/ledger/internal/state/state_migration_v3.cc (hunk 1 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/state/state_migration_v3.cc:6]: done

File: vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.h (hunk 2 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.h:89]: done

File: components/brave_rewards/browser/external_wallet.cc [hunk 3]
```diff
@@ -15,24 +15,14 @@ namespace brave_rewards {
...
-    status = properties.status;
-    type = properties.type;
-    verify_url = properties.verify_url;
-    add_url = properties.add_url;
-    withdraw_url = properties.withdraw_url;
-    user_name = properties.user_name;
-    account_url = properties.account_url;
-    login_url = properties.login_url;
-  }
+  ExternalWallet::ExternalWallet(const ExternalWallet &properties) =  default;
```
[the remark this answers is not in the document]
Comment (NejcZdovc) [reply components/brave_rewards/browser/external_wallet.cc:18]: fixed

**Comment by NejcZdovc** [comment]
CI failed on test-install

**Pull request merged**